client/src/annoMatrix/annoMatrix.ts

```diff
@@ -10,45 +10,65 @@ import {
   _getWritableColumns,
 } from "./schema";
 import { indexEntireSchema } from "../util/stateManager/schemaHelpers";
-import { _whereCacheGet, _whereCacheMerge } from "./whereCache";
+import { _whereCacheGet, _whereCacheMerge, WhereCache } from "./whereCache";
 import _shallowClone from "./clone";
-import { _queryValidate, _queryCacheKey } from "./query";
+import { _queryValidate, _queryCacheKey, Query } from "./query";
+import {
+  AnnotationColumn,
+  Category,
+  DataFrame,
+  Field,
+  GCHints,
+  LayoutColumn,
+  Schema,
+} from "../common/types/entities";
 
 const _dataframeCache = dataframeMemo(128);
 
-export default class AnnoMatrix {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  public isView: any;
+interface Cache {
+  [Field.obs]: Dataframe;
+  [Field.var]: Dataframe;
+  [Field.emb]: Dataframe;
+  [Field.X]: Dataframe;
+}
+
+export type ObsColumnValue = Category | null | undefined;
+
+interface PendingLoad {
+  [Field.obs]: { [key: string]: Promise<void> };
+  [Field.var]: { [key: string]: Promise<void> };
+  [Field.emb]: { [key: string]: Promise<void> };
+  [Field.X]: { [key: string]: Promise<void> };
+}
+
+export interface UserFlags {
+  isUserSubsetView?: boolean;
+  isEmbSubsetView?: boolean;
+}
+
+export default abstract class AnnoMatrix {
```

Comment:
Modified `AnnoMatrix` to be abstract, as well as its `addObsAnnoCategory`, `removeObsAnnoCategory`, `dropObsColumn`, `addObsColumn`, `renameObsColumn`, `setObsColumnValues`, `resetObsColumnValues` and `addEmbedding` methods.